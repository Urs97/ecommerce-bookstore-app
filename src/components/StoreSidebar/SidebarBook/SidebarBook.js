import React from 'react';
import './SidebarBook.scss';

const SidebarBook = ({ book }) => {

    // L is large, M is medium and S is small image format
    const img_url = `http://covers.openlibrary.org/b/olid/${book.cover_edition_key}-L.jpg`;
    const img_alt = `${book.title} book cover`;

    // To add link functionality to book

    return (
        <section className="sidebar-book" tabIndex="0">
            <img src={img_url} alt={img_alt} className="sidebar-book-img"/>
            <section className="sidebar-book-details">
                <h4 className="title">{book.title}</h4>
                <p>{book.authors[0].name}</p>
                <span className="sidebar-book-price-container">${book.price[0]}.{book.price[1]}</span>
            </section>
        </section>
    )
};

export default SidebarBook;
