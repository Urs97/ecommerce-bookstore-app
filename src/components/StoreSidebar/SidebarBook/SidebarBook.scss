@import '../../../styles/shared.scss';

.sidebar-book {
    display: flex;
    color: $text-color;
    cursor: pointer;

    .sidebar-book-img {
        width: 5em;
        height: 7em;
        background-color: $blue;
        border-radius: $border-radius;
    }

    .sidebar-book-details { 
        padding: 0.5em 1em;
        display: flex;
        flex-direction: column;
        justify-content: center;

        h4 {
            font-size: 0.95em;
            overflow: hidden;
            // Triple native line height of 1.15em
            max-height: 3.45em;
        }

        p {
            font-size: 0.8em;
        }

        .sidebar-book-price-container {
            font-size: 0.95em;
        }
    }

    .sidebar-book-details > *:not(:last-child) {
        margin-bottom: 0.5em;
    }
}

    .sidebar-book:focus {
        outline: 0.2em solid black;
    }

    .sidebar-book:not(.focus-visible) {
        outline: none;
    }

@media (max-width: 800px) {
    .sidebar-book {
        .sidebar-book-details {
            flex-grow: 1;
            display: grid;
            grid-template-areas: 
            "title"
            "author"
            "price";
            grid-template-rows: 3fr 1fr 1fr;

            h4 {
                grid-area: title;
                align-self: center;
            }

            p {
                grid-area: author;
                align-self: center;
            }
    
            .sidebar-book-price-container {
                grid-area: price;
            }
        }

        .sidebar-book-details > * {
            text-align: center;
        }
    }
}