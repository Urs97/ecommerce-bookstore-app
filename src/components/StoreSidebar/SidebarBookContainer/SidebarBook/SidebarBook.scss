@import '../../../../styles/shared.scss';

.sidebar-book {
    display: flex;
    color: $text-color;

    .sidebar-book-img {
        width: 5em;
        height: 7em;
        background-color: $blue;
        border-radius: $border-radius;
    }

    .sidebar-book-details { 
        padding: 0.5em 1em;
        display: flex;
        flex-direction: column;
        justify-content: center;

        .sidebar-book-title-container {
            font-size: 0.95em;
        }

        p {
            font-size: 0.8em;
        }

        .sidebar-book-price-container {
            font-size: 0.95em;
        }
    }

    .sidebar-book-details > *:not(:last-child) {
        margin-bottom: 0.5em;
    }
}