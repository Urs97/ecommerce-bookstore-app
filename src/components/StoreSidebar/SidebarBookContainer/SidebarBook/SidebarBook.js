import React from 'react';
import './SidebarBook.scss';

function SidebarBook({ book }) {

    // L is large, M is medium and S is small image format
    const img_url = `http://covers.openlibrary.org/b/olid/${book.cover_edition_key}-L.jpg`;
    const img_alt = `${book.title} book cover`;

    return (
        <section className="sidebar-book">
            <img src={img_url} alt={img_alt} className="sidebar-book-img"/>
            <div className="sidebar-book-details">
                <h4 className="sidebar-book-title-container title">{book.title}</h4>
                <p>{book.authors[0].name}</p>
                <span className="sidebar-book-price-container">${book.price[0]}.{book.price[1]}</span>
            </div>
        </section>
    )
};

export default SidebarBook;
