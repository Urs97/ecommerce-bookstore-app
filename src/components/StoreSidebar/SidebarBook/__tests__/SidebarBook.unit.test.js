import React from 'react';
import {shallow} from 'enzyme';
import SidebarBook from '../SidebarBook';

describe('SidebarBook', () => {
    test('renders empty SidebarBook', () => {
        const wrapper = shallow(
            <SidebarBook />
        );
        expect(wrapper).toMatchSnapshot();
    });
});