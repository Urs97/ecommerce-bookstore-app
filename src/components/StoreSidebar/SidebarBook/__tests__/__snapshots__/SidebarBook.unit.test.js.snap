// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarBook renders SidebarBook 1`] = `
<section
  className="sidebar-book"
>
  <img
    alt="Frankenstein book cover"
    className="sidebar-book-img"
    src="http://covers.openlibrary.org/b/olid/OL6147070M-L.jpg"
  />
  <section
    className="sidebar-book-details"
  >
    <h4
      className="title"
    >
      Frankenstein
    </h4>
    <p>
      Marry Shelley
    </p>
    <span
      className="sidebar-book-price-container"
    >
      $
      94
      .
      99
    </span>
  </section>
</section>
`;
